### PR TITLE
chore: Improve documentations

### PR DESCRIPTION
**Pull request by tianfeng92**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
   platformName: "Windows 11"
 });
 console.log(job.id); // the job ID
 console.log(job.url); // the full URL of the job
 ```
 
-## Upload Assets
+## Upload log for JavaScript Batch Frameworks
+
+The minimum `sauce-test-report.json` file to fullfill the web UI:
```
Comment (alexplischke) [inline README.md:40]: Typo, but also, what does "fulfill the web UI" mean?
```suggestion
The minimum `sauce-test-report.json` file to fulfill the web UI:
```

File: README.md [hunk 2]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
+```
+To upload `sauce-test-report.json`:
 
 ```javascript
-const Readable = require('stream').Readable;
+const uploads = await client.uploadAssets(
+  job.id,
+  [{
+    filename: "log.json",
+    data: fs.createReadStream("log.json")
```
Comment (alexplischke) [inline README.md:84]: Aren't you [trying to upload](https://github.com/saucelabs/node-testcomposer/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R77) "sauce-test-report.json"?

File: README.md [hunk 3]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
+  framework: "playwright", // alternatives: cypress, playwright, testcafe
   frameworkVersion: "1.25.0",
   platformName: "Windows 11"
 });
 console.log(job.id); // the job ID
 console.log(job.url); // the full URL of the job
 ```
 
-## Upload Assets
+## Upload log for JavaScript Batch Frameworks
```
Comment (alexplischke) [inline README.md:38]: Imho this should be a subtopic under https://github.com/saucelabs/node-testcomposer/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R19

File: README.md [hunk 4]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
+  browserName: "Chrome",
+  browserVersion: "105",
+  framework: "webdriver",
+  platformName: "Windows 11"
 });
+console.log(job.id); // the job ID
+console.log(job.url); // the full URL of the job
+```
+
+## Upload Log for Selenium job
```
Comment (alexplischke) [inline README.md:108]: Subtopic 🙏

File: README.md [hunk 5]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
 
-const s = new Readable();
-s.push('hello!');
-s.push(null);
+The minimum `log.json` file to fullfill the web UI:
```
Comment (alexplischke) [inline README.md:110]: Same as above.
```suggestion
The minimum `log.json` file to fulfill the web UI:
```

File: README.md [hunk 6]
```diff
@@ -23,31 +27,121 @@ const job = await client.createReport({
...
+
+    ],
+    "start_time": 1688673799.516,
+    "request": {
+      "using": "css selector",
+      "value": "[id=\"onetrust-accept-btn-handler\"]"
+    },
+    "result": {
+      "message": "no such element",
+      "error": "no such element"
```
Comment (alexplischke) [inline README.md:126]: You say this is the minimal input, but can't result be empty? Wouldn't that make this optional?

Would be also good to add some description as to what these fields mean.
Like `"statusCode": 1`. What values are acceptable?

File: README.md (hunk 2 above)
Comment (tianfeng92) [reply README.md:84]: ah.. Nice catch!

File: README.md (hunk 3 above)
Comment (tianfeng92) [reply README.md]: fixed 👍

File: README.md [hunk 7]
```diff
@@ -4,17 +4,21 @@ Sauce Labs SDK for the Test-Composer API.
 
 # Usage
 
-## Create Reports
-
+## Initial TestComposer Client
```
Comment (mhan83) [inline README.md:7]:
```suggestion
## Initialize TestComposer Client
```

File: README.md [hunk 8]
```diff
@@ -4,17 +4,21 @@ Sauce Labs SDK for the Test-Composer API.
...
 
 const client = new TestComposer({
   region: 'us-west-1',
   username: process.env.SAUCE_USERNAME,
   accessKey: process.env.SAUCE_ACCESS_KEY,
   headers: {'User-Agent': `your-fancy-reporter/1.2.3`}
 });
+```
+
+## Create Reports for JavaScript Batch Frameworks
```
Comment (mhan83) [inline README.md:19]: Is referring to "Batch Frameworks" here helpful? "Batch" is a name we use internally and probably doesn't connote anything meaningful to most users.

File: README.md [hunk 9]
```diff
@@ -23,31 +27,117 @@ const job = await client.createReport({
...
   platformName: "Windows 11"
 });
 console.log(job.id); // the job ID
 console.log(job.url); // the full URL of the job
 ```
 
-## Upload Assets
+### Upload log for JavaScript Batch Frameworks
+
+The minimum `sauce-test-report.json` file to fulfill the web UI:
```
Comment (mhan83) [inline README.md:40]: "Fulfill the web UI" is a bit terse and awkward. Can you describe a bit more clearly what the file is for?

And the phrasing `The minimum 'sauce-test-report.json' file` makes it sound like the contents of the snippet verbatim below is the "minimum" report required. But I'm guessing what you mean is at a minimum, a file called `sauce-test-report.json` is required?

Should we then also refer to the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) lib for generating the reports?

File: README.md (hunk 8 above)
Comment (tianfeng92) [reply README.md:19]: What about just `JavaScript Frameworks`?

File: README.md [hunk 10]
```diff
@@ -23,31 +27,130 @@ const job = await client.createReport({
...
   platformName: "Windows 11"
 });
 console.log(job.id); // the job ID
 console.log(job.url); // the full URL of the job
 ```
 
-## Upload Assets
+### Upload Log
+
+The minimum `sauce-test-report.json` file, which can be generated by [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js/), to populate the test case list on Sauce Labs web UI:
```
Comment (mhan83) [inline README.md:40]:
```suggestion
In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).
```
```

**Review by mhan83 — APPROVED** [review]

File: README.md [hunk 11]
```diff
@@ -23,31 +27,130 @@ const job = await client.createReport({
...
   platformName: "Windows 11"
 });
 console.log(job.id); // the job ID
 console.log(job.url); // the full URL of the job
 ```
 
-## Upload Assets
+### Upload Log
+
+In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).
```
Comment (alexplischke) [inline README.md:40]:
```suggestion
In order to view your test results in the SauceLabs web app, you must upload a test report file along with the report created in the step above. You can use the [sauce-json-reporter](https://github.com/saucelabs/sauce-json-reporter-js) library to generate the test report programmatically or manually create the json file according to the report schema defined [here](https://github.com/saucelabs/sauce-json-reporter-js/blob/main/api/schema.json).
```

Comment (mhan83) [reply README.md]: Huh, I've been misspelling that my whole career, apparently. 🤦

**Review by alexplischke — APPROVED** [review]

**Pull request merged**